asv_bench/benchmarks/utils/common.py

```diff
@@ -289,10 +268,19 @@ def gen_data(
         "str_int": gen_str_int_data,
         "true_false_int": gen_true_false_int_data,
     }
+    cache_key = (data_type, nrows, ncols, rand_low, rand_high)
+    if cache_key in data_cache:
+        return data_cache[cache_key]
+
+    logging.info(
+        "Generating {} data {} rows and {} columns [{}-{}]".format(
+            data_type, nrows, ncols, rand_low, rand_high
+        )
+    )
     assert data_type in type_to_generator
     data_generator = type_to_generator[data_type]
 
-    data = data_generator(nrows, ncols, rand_low, rand_high)
+    data = data_generator(nrows, ncols, rand_low, rand_high, cache_key)
```

Comment:
and remove that new line from the lower level functions?
```suggestion
    data = data_generator(nrows, ncols, rand_low, rand_high, cache_key)
    data_cache[cache_key] = weakdict(data)
```

Reply:
Done